projects/components/alerts/alerts.service.ts

```diff
@@ -440,14 +441,14 @@ export class AlertsService implements OnDestroy {
      * @returns a boolean promise resolved when the user closes the dialog
      * the result is true is the list was updated.
      */
-    public manageAlertsModal() : Promise<boolean> {
+    public manageAlertsModal(searchRoute?: string) : Promise<boolean> {
 
-        const model: ManageAlertsModel = { alerts: Utils.copy(this.alerts) };
+        const model: { alertManager: ManageAlertsModel, searchRoute?: string } = { alertManager: { alerts: Utils.copy(this.alerts) }, searchRoute: searchRoute };
```

Comment:
Rather than nesting ManageAlertsModel within a wrapper, could you add `searchRoute` to the `ManageAlertsModel` interface?

Reply:
You would have simply:
`const model: ManageAlertsModel = { alerts: Utils.copy(this.alerts), searchRoute }`

Reply:
Sure, will add this change.